About the temporal discounting task.

In [1]:
import numpy as np
import pandas as pd

From Cohen's thesis:

> For the temporal discounting task, the model included events for trials
> predefined as hard choices and trials predefined as easy choices. It also
> included parametric regressors weighting each of those trial types by delay.
> All events began at stimulus onset. The duration of all events was the
> participant's RT on that trial. Nuisance regressors included trials with no
> response (duration 4.5 seconds) and the remainder of each trial itself after
> a decision had been made (4.5 seconds - RT).

In [2]:
df = pd.read_table('ds000009_R2.0.3/sub-02/func/sub-02_task-discounting_events.tsv')
df

onset  duration  reaction_time  percent_off_trial trial_type  \
0    16.01       4.5         1.2609               1.90   easy_par   
1    21.25       4.5         1.5137              -0.95   easy_par   
2    31.76       4.5         1.2677               2.00   easy_par   
3    37.00       4.5         1.0086              -0.99   easy_par   
4    42.26       4.5         1.2752              -0.01   hard_par   
5    47.51       4.5         0.9462               1.90   easy_par   
6    52.75       4.5         0.9588              -0.95   easy_par   
7    58.01       4.5         0.9581              -0.99   easy_par   
8    63.26       4.5         0.9246              -0.05   hard_par   
9    68.52       4.5         0.6006              -0.05   hard_par   
10   73.76       4.5         0.6805              -0.01   hard_par   
11   79.01       4.5         0.7096               1.90   easy_par   
12   84.27       4.5         1.1926               2.00   easy_par   
13   89.51       4.5         1.1318               2.00   easy_par   
14   94.76       4.5         1.3282               2.00   easy_par   
15  105.26       4.5         0.9451              -0.01   hard_par   
16  110.50       4.5         0.6604               0.01   hard_par   
17  115.77       4.5         0.7188               0.05   hard_par   
18  121.01       4.5         0.4976               1.90   easy_par   
19  126.25       4.5         0.6136              -0.01   hard_par   
20  131.51       4.5         0.5385               0.01   hard_par   
21  136.76       4.5         0.5068              -0.95   easy_par   
22  142.00       4.5         0.5922               2.00   easy_par   
23  147.26       4.5         0.9492               0.05   hard_par   
24  152.51       4.5         0.5512               2.00   easy_par   
25  157.75       4.5         0.5533               0.01   hard_par   
26  163.01       4.5         0.6045               0.05   hard_par   
27  168.26       4.5         0.9610               0.05   hard_par   
28  178.76       4.5         0.7048              -0.95   easy_par   
29  184.01       4.5         0.9019               0.05   hard_par   
..     ...       ...            ...                ...        ...   
66  388.77       4.5         0.9248              -0.05   hard_par   
67  399.25       4.5         1.0750              -0.05   hard_par   
68  404.52       4.5         1.1980               1.90   easy_par   
69  409.76       4.5         0.8426              -0.01   hard_par   
70  415.00       4.5         0.7256              -0.01   hard_par   
71  420.27       4.5         0.6179               0.05   hard_par   
72  425.51       4.5         0.6606               2.00   easy_par   
73  430.75       4.5         0.8481              -0.01   hard_par   
74  436.01       4.5         1.3652               0.05   hard_par   
75  441.26       4.5         1.4551               1.90   easy_par   
76  446.50       4.5         1.3411              -0.01   hard_par   
77  451.76       4.5         1.0144               0.01   hard_par   
78  457.01       4.5         0.7166               2.00   easy_par   
79  462.25       4.5         0.6409              -0.95   easy_par   
80  472.76       4.5         0.8857              -0.95   easy_par   
81  478.00       4.5         0.5783               0.01   hard_par   
82  483.26       4.5         0.7430               0.01   hard_par   
83  488.51       4.5         1.2069              -0.99   easy_par   
84  493.75       4.5         0.9238              -0.01   hard_par   
85  499.01       4.5         1.7584               1.90   easy_par   
86  504.26       4.5         0.7779              -0.99   easy_par   
87  509.52       4.5         1.2203               1.90   easy_par   
88  514.76       4.5         0.0000               0.05   hard_par   
89  520.01       4.5         1.5107               2.00   easy_par   
90  525.27       4.5         1.0814              -0.05   hard_par   
91  530.51       4.5         0.8362               0.01   hard_par   
92 

In [3]:
df.columns

Index(['onset', 'duration', 'reaction_time', 'percent_off_trial', 'trial_type',
       'immediate_amount', 'delayed_amount', 'delay_time_words',
       'delay_time_days', 'k_this_trial', 'response_button',
       'response_binarized', 'onset.plus.rt', 'onset.plus.stimdur',
       'absolute_time', 'trialnum', 'onset_orig', 'duration_orig',
       'trial_type_orig', 'easy_par_orig', 'hard_par_orig'],
      dtype='object')

Oddly, the `onset` field appears to be a rounded version of the `onset_orig` field:

In [4]:
assert np.all(df['onset'] == df['onset_orig'].round(2))

Old onsets `cond001.txt` is `easy` trial, with standard duration.

In [5]:
cat old_onsets/sub002/model/model001/onsets/task004_run001/cond001.txt

16.0082	4.5	1
21.2524	4.5	1
31.7574	4.5	1
37.0017	4.5	1
47.5068	4.5	1
52.7511	4.5	1
58.0121	4.5	1
79.0056	4.5	1
84.2666	4.5	1
89.5109	4.5	1
94.7551	4.5	1
121.0097	4.5	1
136.7590	4.5	1
142.0036	4.5	1
152.5085	4.5	1
178.7631	4.5	1
194.5125	4.5	1
199.7613	4.5	1
226.0112	4.5	1
231.2554	4.5	1
241.7607	4.5	1
252.2659	4.5	1
257.5101	4.5	1
268.0153	4.5	1
273.2595	4.5	1
283.7648	4.5	1
289.0090	4.5	1
304.7584	4.5	1
310.0027	4.5	1
325.7519	4.5	1
331.0128	4.5	1
352.0067	4.5	1
362.5117	4.5	1
373.0168	4.5	1
378.2611	4.5	1
383.5053	4.5	1
404.5157	4.5	1
425.5093	4.5	1
441.2588	4.5	1
457.0082	4.5	1
462.2525	4.5	1
472.7576	4.5	1
488.5071	4.5	1
499.0121	4.5	1
504.2564	4.5	1
509.5172	4.5	1
520.0059	4.5	1
535.7552	4.5	1


`cond002.txt` is `easy` with parametric modulation by delay.

In [6]:
cat old_onsets/sub002/model/model001/onsets/task004_run001/cond002.txt

16.0082	4.5	-0.0625
21.2524	4.5	-0.0625
31.7574	4.5	-0.0625
37.0017	4.5	-0.0625
47.5068	4.5	-0.0625
52.7511	4.5	-0.0625
58.0121	4.5	-0.0625
79.0056	4.5	-0.0625
84.2666	4.5	-0.0625
89.5109	4.5	-0.0625
94.7551	4.5	-0.0625
121.0097	4.5	-0.0625
136.7590	4.5	-0.0625
142.0036	4.5	-0.0625
152.5085	4.5	-0.0625
178.7631	4.5	-0.0625
194.5125	4.5	-0.0625
199.7613	4.5	-0.0625
226.0112	4.5	-0.0625
231.2554	4.5	-0.0625
241.7607	4.5	-0.0625
252.2659	4.5	-0.0625
257.5101	4.5	-0.0625
268.0153	4.5	-0.0625
273.2595	4.5	-0.0625
283.7648	4.5	-0.0625
289.0090	4.5	-0.0625
304.7584	4.5	-0.0625
310.0027	4.5	-0.0625
325.7519	4.5	-0.0625
331.0128	4.5	-0.0625
352.0067	4.5	-0.0625
362.5117	4.5	-0.0625
373.0168	4.5	-0.0625
378.2611	4.5	-0.0625
383.5053	4.5	-0.0625
404.5157	4.5	-0.0625
425.5093	4.5	-0.0625
441.2588	4.5	0.9375
457.0082	4.5	-0.0625
462.2525	4.5	-0.0625
472.7576	4.5	-0.0625
488.5071	4.5	-0.0625
499.0121	4.5	0.9375
504.2564	4.5	-0.0625
509.5172	4.5	-0.0625
520.0059	4.5	0.9375
535.7552	4.5	-0.0625


`cond003` is hard, with standard duration:

In [7]:
cat old_onsets/sub002/model/model001/onsets/task004_run001/cond003.txt

42.2626	4.5	1
63.2562	4.5	1
68.5172	4.5	1
73.7614	4.5	1
105.2602	4.5	1
110.5044	4.5	1
115.7654	4.5	1
126.2540	4.5	1
131.5148	4.5	1
147.2643	4.5	1
157.7527	4.5	1
163.0137	4.5	1
168.2579	4.5	1
184.0073	4.5	1
189.2516	4.5	1
205.0009	4.5	1
210.2620	4.5	1
215.5061	4.5	1
220.7669	4.5	1
236.5163	4.5	1
262.7544	4.5	1
278.5036	4.5	1
294.2532	4.5	1
299.5141	4.5	1
315.2635	4.5	1
336.2571	4.5	1
341.5014	4.5	1
346.7624	4.5	1
357.2508	4.5	1
367.7560	4.5	1
388.7663	4.5	1
399.2548	4.5	1
409.7599	4.5	1
415.0042	4.5	1
420.2652	4.5	1
430.7537	4.5	1
436.0145	4.5	1
446.5031	4.5	1
451.7640	4.5	1
478.0018	4.5	1
483.2628	4.5	1
493.7513	4.5	1
514.7615	4.5	1
525.2667	4.5	1
530.5110	4.5	1
546.2604	4.5	1
551.5047	4.5	1
556.7656	4.5	1


`cond004` is the parametric modulation of `hard`:

In [8]:
cat old_onsets/sub002/model/model001/onsets/task004_run001/cond004.txt

42.2626	4.5	-2.0833
63.2562	4.5	-2.0833
68.5172	4.5	-2.0833
73.7614	4.5	-2.0833
105.2602	4.5	-2.0833
110.5044	4.5	-2.0833
115.7654	4.5	-2.0833
126.2540	4.5	-2.0833
131.5148	4.5	-2.0833
147.2643	4.5	-2.0833
157.7527	4.5	-2.0833
163.0137	4.5	-2.0833
168.2579	4.5	-2.0833
184.0073	4.5	-2.0833
189.2516	4.5	-2.0833
205.0009	4.5	-2.0833
210.2620	4.5	-2.0833
215.5061	4.5	-2.0833
220.7669	4.5	-2.0833
236.5163	4.5	-2.0833
262.7544	4.5	-2.0833
278.5036	4.5	-2.0833
294.2532	4.5	-2.0833
299.5141	4.5	-2.0833
315.2635	4.5	-2.0833
336.2571	4.5	-2.0833
341.5014	4.5	-2.0833
346.7624	4.5	-2.0833
357.2508	4.5	-2.0833
367.7560	4.5	-2.0833
388.7663	4.5	-2.0833
399.2548	4.5	-2.0833
409.7599	4.5	-2.0833
415.0042	4.5	-2.0833
420.2652	4.5	-2.0833
430.7537	4.5	-2.0833
436.0145	4.5	-2.0833
446.5031	4.5	-2.0833
451.7640	4.5	-2.0833
478.0018	4.5	-2.0833
483.2628	4.5	-2.0833
493.7513	4.5	-2.0833
514.7615	4.5	96.9167
525.2667	4.5	-2.0833
530.5110	4.5	-2.0833
546.2604	4.5	-2.0833
551.5047	4.5	-2.0833
556.7656	4.5	-1.0

But - there's something wrong - the onset starting 514 is a trial with no response - and the parametric modulator looks off.  As we'll see in a bit, the 96.9 value comes from innocently re-using the 99 value that flags the lack of response.

`cond005` is trials with no response.

In [9]:
cat old_onsets/sub002/model/model001/onsets/task004_run001/cond005.txt

514.7615	4.5	1


In [10]:
easy_par_numeric = pd.to_numeric(df['easy_par_orig'], 'coerce')
easy_par_numeric

0    -0.0625
1    -0.0625
2    -0.0625
3    -0.0625
4        NaN
5    -0.0625
6    -0.0625
7    -0.0625
8        NaN
9        NaN
10       NaN
11   -0.0625
12   -0.0625
13   -0.0625
14   -0.0625
15       NaN
16       NaN
17       NaN
18   -0.0625
19       NaN
20       NaN
21   -0.0625
22   -0.0625
23       NaN
24   -0.0625
25       NaN
26       NaN
27       NaN
28   -0.0625
29       NaN
       ...  
66       NaN
67       NaN
68   -0.0625
69       NaN
70       NaN
71       NaN
72   -0.0625
73       NaN
74       NaN
75    0.9375
76       NaN
77       NaN
78   -0.0625
79   -0.0625
80   -0.0625
81       NaN
82       NaN
83   -0.0625
84       NaN
85    0.9375
86   -0.0625
87   -0.0625
88       NaN
89    0.9375
90       NaN
91       NaN
92   -0.0625
93       NaN
94       NaN
95       NaN
Name: easy_par_orig, dtype: float64

The parametric scalings don't come from the delay, as Cohen describes.   There are only two parametric scaling values:

In [11]:
np.unique(easy_par_numeric.dropna())

array([-0.0625,  0.9375])

Here are the only three rows generating the positive scaling values:

In [12]:
df[easy_par_numeric > 0]

onset  duration  reaction_time  percent_off_trial trial_type  \
75  441.26       4.5         1.4551                1.9   easy_par   
85  499.01       4.5         1.7584                1.9   easy_par   
89  520.01       4.5         1.5107                2.0   easy_par   

    immediate_amount  delayed_amount delay_time_words  delay_time_days  \
75                 5           6.588           1 year              365   
85                10          11.044         4 months              120   
89                 5           6.643           1 year              365   

    k_this_trial      ...      response_binarized  onset.plus.rt  \
75       0.00087      ...                       1         442.71   
85       0.00087      ...                       1         500.77   
89       0.00090      ...                       1         521.52   

    onset.plus.stimdur  absolute_time  trialnum  onset_orig  duration_orig  \
75              445.77         425.25        82    441.2588            4.5   
85              503.53         483.00        93    499.0121            4.5   
89              524.52         504.00        97    520.0059            4.5   

    trial_type_orig easy_par_orig hard_par_orig  
75         easy_par        0.9375           n/a  
85         easy_par        0.9375           n/a  
89         easy_par        0.9375           n/a  

[3 rows x 21 columns]

In contrast, here are the rows with the negative scaling values.

In [13]:
df[easy_par_numeric < 0]

onset  duration  reaction_time  percent_off_trial trial_type  \
0    16.01       4.5         1.2609               1.90   easy_par   
1    21.25       4.5         1.5137              -0.95   easy_par   
2    31.76       4.5         1.2677               2.00   easy_par   
3    37.00       4.5         1.0086              -0.99   easy_par   
5    47.51       4.5         0.9462               1.90   easy_par   
6    52.75       4.5         0.9588              -0.95   easy_par   
7    58.01       4.5         0.9581              -0.99   easy_par   
11   79.01       4.5         0.7096               1.90   easy_par   
12   84.27       4.5         1.1926               2.00   easy_par   
13   89.51       4.5         1.1318               2.00   easy_par   
14   94.76       4.5         1.3282               2.00   easy_par   
18  121.01       4.5         0.4976               1.90   easy_par   
21  136.76       4.5         0.5068              -0.95   easy_par   
22  142.00       4.5         0.5922               2.00   easy_par   
24  152.51       4.5         0.5512               2.00   easy_par   
28  178.76       4.5         0.7048              -0.95   easy_par   
31  194.51       4.5         1.6501               1.90   easy_par   
32  199.76       4.5         0.9463              -0.99   easy_par   
37  226.01       4.5         1.1998              -0.99   easy_par   
38  231.26       4.5         1.1384               2.00   easy_par   
40  241.76       4.5         0.7322              -0.95   easy_par   
41  252.27       4.5         0.8637              -0.95   easy_par   
42  257.51       4.5         0.7800              -0.95   easy_par   
44  268.02       4.5         1.1867               1.90   easy_par   
45  273.26       4.5         0.7827              -0.99   easy_par   
47  283.76       4.5         1.2301               2.00   easy_par   
48  289.01       4.5         0.5861              -0.99   easy_par   
51  304.76       4.5         0.9354              -0.99   easy_par   
52  310.00       4.5         1.5916               1.90   easy_par   
54  325.75       4.5         0.8289              -0.95   easy_par   
55  331.01       4.5         0.6713              -0.99   easy_par   
59  352.01       4.5         0.8773              -0.95   easy_par   
61  362.51       4.5         0.7025              -0.99   easy_par   
63  373.02       4.5         1.1150              -0.95   easy_par   
64  378.26       4.5         1.3113               2.00   easy_par   
65  383.51       4.5         0.9137              -0.99   easy_par   
68  404.52       4.5         1.1980               1.90   easy_par   
72  425.51       4.5         0.6606               2.00   easy_par   
78  457.01       4.5         0.7166               2.00   easy_par   
79  462.25       4.5         0.6409              -0.95   easy_par   
80  472.76       4.5         0.8857              -0.95   easy_par   
83  488.51       4.5         1.2069              -0.99   easy_par   
86  504.26       4.5         0.7779              -0.99   easy_par   
87  509.52       4.5         1.2203               1.90   easy_par   
92  535.76       4.5         0.6169               1.90   easy_par   

    immediate_amount  delayed_amount delay_time_words  delay_time_days  \
0                 10          10.183          3 weeks               21   
1                 10          10.000           2 days                2   
2                  5           5.540         4 months              120   
3                  5           5.005           1 year              365   
5                  5           5.030           1 week                7   
6                 10          10.055           1 year              365   
7                  5           5.001         2 months               60   
11                10          10.017           2 days                2   
12                 5           5.095          3 weeks               21   
13                10          10.189          3 weeks               21   
14              

How about correlations?

In [14]:
df['easy_par_numeric'] = easy_par_numeric
df.corr()

onset  duration  reaction_time  percent_off_trial  \
onset               1.000000       NaN       0.142923      -5.425410e-02   
duration                 NaN       NaN            NaN                NaN   
reaction_time       0.142923       NaN       1.000000       2.493771e-01   
percent_off_trial  -0.054254       NaN       0.249377       1.000000e+00   
immediate_amount   -0.113093       NaN      -0.006061       1.045706e-17   
delayed_amount     -0.105025       NaN       0.078239       8.648854e-02   
delay_time_days     0.000037       NaN       0.258964      -2.142579e-17   
k_this_trial       -0.054254       NaN       0.249377       1.000000e+00   
response_binarized  0.155203       NaN      -0.284944      -1.409348e-02   
onset.plus.rt       0.944518       NaN       0.240734      -4.713287e-02   
onset.plus.stimdur  1.000000       NaN       0.142924      -5.425985e-02   
absolute_time       1.000000       NaN       0.142922      -5.426011e-02   
trialnum            1.000000       NaN       0.142922      -5.426011e-02   
onset_orig          1.000000       NaN       0.142924      -5.425792e-02   
duration_orig            NaN       NaN            NaN                NaN   
easy_par_numeric    0.356266       NaN       0.464843       2.551646e-01   

                    immediate_amount  delayed_amount  delay_time_days  \
onset                  -1.130931e-01       -0.105025     3.651546e-05   
duration                         NaN             NaN              NaN   
reaction_time          -6.060731e-03        0.078239     2.589637e-01   
percent_off_trial       1.045706e-17        0.086489    -2.142579e-17   
immediate_amount        1.000000e+00        0.978304     0.000000e+00   
delayed_amount          9.783035e-01        1.000000     1.342946e-01   
delay_time_days         0.000000e+00        0.134295     1.000000e+00   
k_this_trial            1.772912e-18        0.086489     5.365031e-17   
response_binarized      1.026212e-01        0.095549    -2.218496e-02   
onset.plus.rt          -1.464080e-01       -0.135300     1.019850e-02   
onset.plus.stimdur     -1.130931e-01       -0.105025     3.659755e-05   
absolute_time          -1.130933e-01       -0.105026     3.759311e-05   
trialnum               -1.130933e-01       -0.105026     3.759311e-05   
onset_orig             -1.130920e-01       -0.105024     3.566083e-05   
duration_orig                    NaN             NaN              NaN   
easy_par_numeric       -8.606630e-02        0.025837     3.815130e-01   

                    k_this_trial  response_binarized  onset.plus.rt  \
onset              -5.425410e-02            0.155203       0.944518   
duration                     NaN                 NaN            NaN   
reaction_time       2.493771e-01           -0.284944       0.240734   
percent_off_trial   1.000000e+00           -0.014093      -0.047133   
immediate_amount    1.772912e-18            0.102621      -0.146408   
delayed_amount      8.648854e-02            0.095549      -0.135300   
delay_time_days     5.365031e-17           -0.022185       0.010199   
k_this_trial        1.000000e+00           -0.014093      -0.047133   
response_binarized -1.409348e-02            1.000000      -0.177810   
onset.plus.rt      -4.713287e-02           -0.177810       1.000000   
onset.plus.stimdur -5.425985e-02            0.155206       0.944517   
absolute_time      -5.426011e-02            0.155202       0.944518   
trialnum           -5.426011e-02            0.155202       0.944518   
onset_orig         -5.425792e-02            0.155204       0.944518   
duration_orig                NaN                 NaN            NaN   
easy_par_numeric    2.551646e-01            1.000000       0.357194   

                    onset.plus.stimdur  absolute_time  trialnum  onset_orig  \
onset                         1.000000       1.000000  1.000000    1.000000   
duration                           NaN            NaN       NaN         NaN   
reaction_time                 0.142924     

Aha - the parameteric regessor comes from `response_binarized` (notice the correlation of 1 with `easy_par_numeric`).

In [15]:
df[['response_button', 'response_binarized', 'easy_par_numeric']]

response_button  response_binarized  easy_par_numeric
0                b                   0           -0.0625
1                b                   0           -0.0625
2                b                   0           -0.0625
3                b                   0           -0.0625
4                b                   0               NaN
5                b                   0           -0.0625
6                b                   0           -0.0625
7                b                   0           -0.0625
8                b                   0               NaN
9                b                   0               NaN
10               b                   0               NaN
11               b                   0           -0.0625
12               b                   0           -0.0625
13               b                   0           -0.0625
14               b                   0           -0.0625
15               b                   0               NaN
16               b                   0               NaN
17               b                   0               NaN
18               b                   0           -0.0625
19               b                   0               NaN
20               b                   0               NaN
21               b                   0           -0.0625
22               b                   0           -0.0625
23               b                   0               NaN
24               b                   0           -0.0625
25               b                   0               NaN
26               b                   0               NaN
27               b                   0               NaN
28               b                   0           -0.0625
29               b                   0               NaN
..             ...                 ...               ...
66               b                   0               NaN
67               b                   0               NaN
68               b                   0           -0.0625
69               b                   0               NaN
70               b                   0               NaN
71               b                   0               NaN
72               b                   0           -0.0625
73               b                   0               NaN
74               b                   0               NaN
75               y                   1            0.9375
76               b                   0               NaN
77               b                   0               NaN
78               b                   0           -0.0625
79               b                   0           -0.0625
80               b                   0           -0.0625
81               b                   0               NaN
82               b                   0               NaN
83               b                   0           -0.0625
84               b                   0               NaN
85               y                   1            0.9375
86               b                   0           -0.0625
87               b                   0           -0.0625
88             n/a                  99               NaN
89               y                   1            0.9375
90               b                   0               NaN
91               b                   0               NaN
92               b                   0           -0.0625
93               b                   0               NaN
94               b                   0               NaN
95               y                   1               NaN

[96 rows x 3 columns]

There are only three responses:

In [16]:
df['response_button'].unique()

array(['b', 'y', 'n/a'], dtype=object)

99 appears to be the missing value flag.

In [17]:
df[df['response_binarized'] == 99]

onset  duration  reaction_time  percent_off_trial trial_type  \
88  514.76       4.5            0.0               0.05   hard_par   

    immediate_amount  delayed_amount delay_time_words  delay_time_days  \
88                10          10.189         2 months               60   

    k_this_trial       ...        onset.plus.rt  onset.plus.stimdur  \
88      0.000315       ...                  0.0              519.28   

    absolute_time  trialnum  onset_orig  duration_orig  trial_type_orig  \
88         498.75        96    514.7615            4.5         hard_par   

    easy_par_orig hard_par_orig easy_par_numeric  
88            n/a       96.9167              NaN  

[1 rows x 22 columns]

The old event files, and the parametric modulator recorded in the `.tsv` file, are therefore not from the delay, as Cohen describes, but from the button press.  Presumably these values don't correspond to the values she used for her thesis.  More on this later on.

For now, let's make the parametric values with the response values here.

In [18]:
TD_RESPONSE_MAP = {'b': 0, 'y': 1, 'n/a': np.nan}

def td_preprocessor_version1(df):
    """ Process dataframe for TD trial types """
    onset_new, duration_orig, trial_type, response, rt, onset_orig = [
        df[name].copy() for name in
        ['onset', 'duration', 'trial_type',
         'response_button', 'reaction_time',
         'onset_orig']]
    # onset_orig has more decimal places, use that instead.  Rounding might in
    # fact be truncation, hence the 0.1 slippage here.
    assert np.max(np.abs(onset_new - onset_orig.round(2))) <= 0.1
    onset_orig.name = 'onset'
    # Use reaction times for duration, for most trials
    rt.name = 'duration'
    # Make responses into numbers
    response = response.map(TD_RESPONSE_MAP)
    tt = trial_type.copy()
    # Use reaction time for duration
    tt[trial_type == 'easy_par'] = 'easy'
    tt[trial_type == 'hard_par'] = 'hard'
    is_missed = response.isnull()
    tt[is_missed] = 'missed'
    # Use original duration (not RT) for missed trials
    rt[is_missed] = duration_orig[is_missed]
    amplitude = pd.Series(np.ones(len(df)), name='amplitude')
    main_trials = pd.concat([tt, onset_orig, rt, amplitude], axis=1)
    # Add parametric trial types
    extras = []
    for t_type in ('easy', 'hard'):
        trial_selector = tt == t_type
        amp_extra = response[trial_selector].copy()
        amp_extra = amp_extra - amp_extra.mean()
        amp_extra.name = 'amplitude'
        tt_extra = tt[trial_selector].copy()
        tt_extra[:] = t_type + 'par'
        extras.append(pd.concat([tt_extra,
                                 onset_orig[trial_selector],
                                 rt[trial_selector],
                                 amp_extra],
                                axis=1))
    # Put the new trials at the end
    return pd.concat([main_trials] + extras, axis=0, ignore_index=True)

In [19]:
ours = td_preprocessor_version1(df)
ours

trial_type     onset  duration  amplitude
0         easy   16.0082    1.2609   1.000000
1         easy   21.2524    1.5137   1.000000
2         easy   31.7574    1.2677   1.000000
3         easy   37.0017    1.0086   1.000000
4         hard   42.2626    1.2752   1.000000
5         easy   47.5068    0.9462   1.000000
6         easy   52.7511    0.9588   1.000000
7         easy   58.0121    0.9581   1.000000
8         hard   63.2562    0.9246   1.000000
9         hard   68.5172    0.6006   1.000000
10        hard   73.7614    0.6805   1.000000
11        easy   79.0056    0.7096   1.000000
12        easy   84.2666    1.1926   1.000000
13        easy   89.5109    1.1318   1.000000
14        easy   94.7551    1.3282   1.000000
15        hard  105.2602    0.9451   1.000000
16        hard  110.5044    0.6604   1.000000
17        hard  115.7654    0.7188   1.000000
18        easy  121.0097    0.4976   1.000000
19        hard  126.2540    0.6136   1.000000
20        hard  131.5148    0.5385   1.000000
21        easy  136.7590    0.5068   1.000000
22        easy  142.0036    0.5922   1.000000
23        hard  147.2643    0.9492   1.000000
24        easy  152.5085    0.5512   1.000000
25        hard  157.7527    0.5533   1.000000
26        hard  163.0137    0.6045   1.000000
27        hard  168.2579    0.9610   1.000000
28        easy  178.7631    0.7048   1.000000
29        hard  184.0073    0.9019   1.000000
..         ...       ...       ...        ...
161    hardpar  215.5061    0.5453  -0.021277
162    hardpar  220.7669    1.2610  -0.021277
163    hardpar  236.5163    0.8941  -0.021277
164    hardpar  262.7544    1.2155  -0.021277
165    hardpar  278.5036    1.3091  -0.021277
166    hardpar  294.2532    0.9337  -0.021277
167    hardpar  299.5141    1.2504  -0.021277
168    hardpar  315.2635    0.8502  -0.021277
169    hardpar  336.2571    0.7553  -0.021277
170    hardpar  341.5014    1.5041  -0.021277
171    hardpar  346.7624    1.4343  -0.021277
172    hardpar  357.2508    1.0149  -0.021277
173    hardpar  367.7560    0.9977  -0.021277
174    hardpar  388.7663    0.9248  -0.021277
175    hardpar  399.2548    1.0750  -0.021277
176    hardpar  409.7599    0.8426  -0.021277
177    hardpar  415.0042    0.7256  -0.021277
178    hardpar  420.2652    0.6179  -0.021277
179    hardpar  430.7537    0.8481  -0.021277
180    hardpar  436.0145    1.3652  -0.021277
181    hardpar  446.5031    1.3411  -0.021277
182    hardpar  451.7640    1.0144  -0.021277
183    hardpar  478.0018    0.5783  -0.021277
184    hardpar  483.2628    0.7430  -0.021277
185    hardpar  493.7513    0.9238  -0.021277
186    hardpar  525.2667    1.0814  -0.021277
187    hardpar  530.5110    0.8362  -0.021277
188    hardpar  546.2604    2.3421  -0.021277
189    hardpar  551.5047    1.2439  -0.021277
190    hardpar  556.7656    1.2387   0.978723

[191 rows x 4 columns]

We aren't getting the same values for `hardpar`, because the original `.tsv` file includes the missed response as a real `hard_par` value, and therefore uses the 99 value, in the calculation of the parametric regressor (which is incorrect).

In [20]:
ours[ours['trial_type'] == 'hardpar']

trial_type     onset  duration  amplitude
144    hardpar   42.2626    1.2752  -0.021277
145    hardpar   63.2562    0.9246  -0.021277
146    hardpar   68.5172    0.6006  -0.021277
147    hardpar   73.7614    0.6805  -0.021277
148    hardpar  105.2602    0.9451  -0.021277
149    hardpar  110.5044    0.6604  -0.021277
150    hardpar  115.7654    0.7188  -0.021277
151    hardpar  126.2540    0.6136  -0.021277
152    hardpar  131.5148    0.5385  -0.021277
153    hardpar  147.2643    0.9492  -0.021277
154    hardpar  157.7527    0.5533  -0.021277
155    hardpar  163.0137    0.6045  -0.021277
156    hardpar  168.2579    0.9610  -0.021277
157    hardpar  184.0073    0.9019  -0.021277
158    hardpar  189.2516    0.5428  -0.021277
159    hardpar  205.0009    0.7731  -0.021277
160    hardpar  210.2620    1.1078  -0.021277
161    hardpar  215.5061    0.5453  -0.021277
162    hardpar  220.7669    1.2610  -0.021277
163    hardpar  236.5163    0.8941  -0.021277
164    hardpar  262.7544    1.2155  -0.021277
165    hardpar  278.5036    1.3091  -0.021277
166    hardpar  294.2532    0.9337  -0.021277
167    hardpar  299.5141    1.2504  -0.021277
168    hardpar  315.2635    0.8502  -0.021277
169    hardpar  336.2571    0.7553  -0.021277
170    hardpar  341.5014    1.5041  -0.021277
171    hardpar  346.7624    1.4343  -0.021277
172    hardpar  357.2508    1.0149  -0.021277
173    hardpar  367.7560    0.9977  -0.021277
174    hardpar  388.7663    0.9248  -0.021277
175    hardpar  399.2548    1.0750  -0.021277
176    hardpar  409.7599    0.8426  -0.021277
177    hardpar  415.0042    0.7256  -0.021277
178    hardpar  420.2652    0.6179  -0.021277
179    hardpar  430.7537    0.8481  -0.021277
180    hardpar  436.0145    1.3652  -0.021277
181    hardpar  446.5031    1.3411  -0.021277
182    hardpar  451.7640    1.0144  -0.021277
183    hardpar  478.0018    0.5783  -0.021277
184    hardpar  483.2628    0.7430  -0.021277
185    hardpar  493.7513    0.9238  -0.021277
186    hardpar  525.2667    1.0814  -0.021277
187    hardpar  530.5110    0.8362  -0.021277
188    hardpar  546.2604    2.3421  -0.021277
189    hardpar  551.5047    1.2439  -0.021277
190    hardpar  556.7656    1.2387   0.978723

In [21]:
df[df['trial_type'] == 'hard_par']['hard_par_orig']

4     -2.0833
8     -2.0833
9     -2.0833
10    -2.0833
15    -2.0833
16    -2.0833
17    -2.0833
19    -2.0833
20    -2.0833
23    -2.0833
25    -2.0833
26    -2.0833
27    -2.0833
29    -2.0833
30    -2.0833
33    -2.0833
34    -2.0833
35    -2.0833
36    -2.0833
39    -2.0833
43    -2.0833
46    -2.0833
49    -2.0833
50    -2.0833
53    -2.0833
56    -2.0833
57    -2.0833
58    -2.0833
60    -2.0833
62    -2.0833
66    -2.0833
67    -2.0833
69    -2.0833
70    -2.0833
71    -2.0833
73    -2.0833
74    -2.0833
76    -2.0833
77    -2.0833
81    -2.0833
82    -2.0833
84    -2.0833
88    96.9167
90    -2.0833
91    -2.0833
93    -2.0833
94    -2.0833
95    -1.0833
Name: hard_par_orig, dtype: object

Notice the 96.9167 at position 88 (I think that's a bug).

After some further exploration, it turns out that some of the old onsets use a different algorithm:

In [22]:
df2 = pd.read_table('ds000009_R2.0.3/sub-13/func/sub-13_task-discounting_events.tsv')
df2

onset  duration  reaction_time  percent_off_trial trial_type  \
0    16.01       4.5         1.4200              -0.99   easy_par   
1    21.26       4.5         2.6965              -0.99   easy_par   
2    31.76       4.5         1.8327               1.90   easy_par   
3    37.01       4.5         1.2259              -0.95   easy_par   
4    42.27       4.5         1.3817              -0.05   hard_par   
5    47.51       4.5         1.3449               0.01   hard_par   
6    52.76       4.5         1.2055               1.90   easy_par   
7    58.02       4.5         1.7193              -0.99   easy_par   
8    63.26       4.5         2.4613              -0.05   hard_par   
9    68.50       4.5         1.2472              -0.01   hard_par   
10   73.77       4.5         1.0420               1.90   easy_par   
11   79.01       4.5         2.3756               0.01   hard_par   
12   84.25       4.5         1.0329               2.00   easy_par   
13   89.51       4.5         2.1665              -0.01   hard_par   
14   94.76       4.5         1.9792               1.90   easy_par   
15  105.26       4.5         1.0478               0.05   hard_par   
16  110.51       4.5         1.2852              -0.05   hard_par   
17  115.75       4.5         1.5847              -0.01   hard_par   
18  121.01       4.5         1.9733              -0.95   easy_par   
19  126.26       4.5         1.4449               0.01   hard_par   
20  131.50       4.5         0.8345               0.01   hard_par   
21  136.76       4.5         1.6846              -0.01   hard_par   
22  142.01       4.5         1.0194              -0.01   hard_par   
23  147.25       4.5         0.7172               0.05   hard_par   
24  152.51       4.5         1.6068              -0.05   hard_par   
25  157.76       4.5         0.8284              -0.05   hard_par   
26  163.00       4.5         1.6713              -0.95   easy_par   
27  168.26       4.5         1.2985              -0.95   easy_par   
28  178.77       4.5         0.9201              -0.95   easy_par   
29  184.01       4.5         0.9003               2.00   easy_par   
..     ...       ...            ...                ...        ...   
66  388.75       4.5         0.7814               1.90   easy_par   
67  399.26       4.5         1.6371              -0.99   easy_par   
68  404.50       4.5         1.5759               0.05   hard_par   
69  409.76       4.5         0.9881               0.05   hard_par   
70  415.01       4.5         0.8154               1.90   easy_par   
71  420.25       4.5         2.1254               1.90   easy_par   
72  425.51       4.5         1.3374               2.00   easy_par   
73  430.76       4.5         0.8680               2.00   easy_par   
74  436.00       4.5         0.5190              -0.99   easy_par   
75  441.26       4.5         0.5781              -0.99   easy_par   
76  446.51       4.5         0.9171              -0.95   easy_par   
77  451.75       4.5         1.1061              -0.05   hard_par   
78  457.01       4.5         0.4685              -0.95   easy_par   
79  462.26       4.5         0.7592              -0.95   easy_par   
80  472.76       4.5         3.4302               0.05   hard_par   
81  478.01       4.5         2.0120               2.00   easy_par   
82  483.27       4.5         0.5962              -0.95   easy_par   
83  488.51       4.5         0.6018              -0.95   easy_par   
84  493.76       4.5         0.8297              -0.05   hard_par   
85  499.02       4.5         0.5352              -0.05   hard_par   
86  504.26       4.5         0.4914               2.00   easy_par   
87  509.50       4.5         0.6789               0.05   hard_par   
88  514.77       4.5         0.7155              -0.05   hard_par   
89  520.01       4.5         1.5576               2.00   easy_par   
90  525.25       4.5         0.8835              -0.99   easy_par   
91  530.51       4.5         0.6068               0.05   hard_par   
92 

Notice that the `duration_orig` values here are not 4.5, as they were for the previous `.tsv`, but do correspond to the reaction time, as Cohen describes.

Notice also that the parametric scaling is now in the hundreds, and doesn't correspond to the response, as it did for the first `.tsv` file we looked at.  Where does that scaling come from?

In [23]:
easy_par_numeric2 = pd.to_numeric(df2['easy_par_orig'], 'coerce')
easy_par_numeric2

0     -35.8333
1     -93.8333
2     -35.8333
3     -93.8333
4          NaN
5          NaN
6     -93.8333
7      24.1667
8          NaN
9          NaN
10     24.1667
11         NaN
12    -74.8333
13         NaN
14    -93.8333
15         NaN
16         NaN
17         NaN
18    -93.8333
19         NaN
20         NaN
21         NaN
22         NaN
23         NaN
24         NaN
25         NaN
26     24.1667
27    -88.8333
28    -35.8333
29    -35.8333
        ...   
66    -74.8333
67    -93.8333
68         NaN
69         NaN
70    -88.8333
71    269.1667
72     24.1667
73    -74.8333
74    -88.8333
75    269.1667
76     24.1667
77         NaN
78    -74.8333
79    269.1667
80         NaN
81    -35.8333
82    -35.8333
83    -74.8333
84         NaN
85         NaN
86    -93.8333
87         NaN
88         NaN
89    269.1667
90    -74.8333
91         NaN
92    -35.8333
93    -88.8333
94    269.1667
95         NaN
Name: easy_par_orig, dtype: float64

In [24]:
df2['easy_par_numeric'] = easy_par_numeric2
df2.corr()

onset  duration  reaction_time  percent_off_trial  \
onset               1.000000       NaN      -0.402182       1.864722e-02   
duration                 NaN       NaN            NaN                NaN   
reaction_time      -0.402182       NaN       1.000000       6.825441e-02   
percent_off_trial   0.018647       NaN       0.068254       1.000000e+00   
immediate_amount   -0.120069       NaN       0.051977      -1.132849e-17   
delayed_amount      0.066096       NaN       0.221567       3.750542e-01   
delay_time_days     0.123600       NaN       0.145883      -6.592550e-18   
k_this_trial        0.018647       NaN       0.068254       1.000000e+00   
response_binarized -0.471651       NaN       0.303281       2.085178e-01   
onset.plus.rt       0.999995       NaN      -0.399259       1.891066e-02   
onset.plus.stimdur  1.000000       NaN      -0.402179       1.864582e-02   
absolute_time       1.000000       NaN      -0.402181       1.865109e-02   
trialnum            1.000000       NaN      -0.402181       1.865109e-02   
onset_orig          1.000000       NaN      -0.402180       1.864726e-02   
duration_orig      -0.402182       NaN       1.000000       6.825441e-02   
easy_par_numeric    0.251940       NaN       0.061884       6.389872e-18   

                    immediate_amount  delayed_amount  delay_time_days  \
onset                  -1.200687e-01        0.066096     1.236001e-01   
duration                         NaN             NaN              NaN   
reaction_time           5.197709e-02        0.221567     1.458830e-01   
percent_off_trial      -1.132849e-17        0.375054    -6.592550e-18   
immediate_amount        1.000000e+00        0.439157     0.000000e+00   
delayed_amount          4.391569e-01        1.000000     5.823563e-01   
delay_time_days         0.000000e+00        0.582356     1.000000e+00   
k_this_trial           -7.294269e-18        0.375054    -1.737248e-17   
response_binarized      2.291746e-01        0.419803     1.323204e-01   
onset.plus.rt          -1.200552e-01        0.066962     1.242818e-01   
onset.plus.stimdur     -1.200685e-01        0.066098     1.236018e-01   
absolute_time          -1.200744e-01        0.066097     1.236025e-01   
trialnum               -1.200744e-01        0.066097     1.236025e-01   
onset_orig             -1.200704e-01        0.066097     1.236014e-01   
duration_orig           5.197709e-02        0.221567     1.458830e-01   
easy_par_numeric       -8.212499e-17        0.555689     1.000000e+00   

                    k_this_trial  response_binarized  onset.plus.rt  \
onset               1.864722e-02           -0.471651       0.999995   
duration                     NaN                 NaN            NaN   
reaction_time       6.825441e-02            0.303281      -0.399259   
percent_off_trial   1.000000e+00            0.208518       0.018911   
immediate_amount   -7.294269e-18            0.229175      -0.120055   
delayed_amount      3.750542e-01            0.419803       0.066962   
delay_time_days    -1.737248e-17            0.132320       0.124282   
k_this_trial        1.000000e+00            0.208518       0.018911   
response_binarized  2.085178e-01            1.000000      -0.471250   
onset.plus.rt       1.891066e-02           -0.471250       1.000000   
onset.plus.stimdur  1.864582e-02           -0.471648       0.999995   
absolute_time       1.865109e-02           -0.471646       0.999995   
trialnum            1.865109e-02           -0.471646       0.999995   
onset_orig          1.864726e-02           -0.471649       0.999995   
duration_orig       6.825441e-02            0.303281      -0.399259   
easy_par_numeric   -2.971481e-17            0.287858       0.252549   

                    onset.plus.stimdur  absolute_time  trialnum  onset_orig  \
onset                         1.000000       1.000000  1.000000    1.000000   
duration                           NaN            NaN       NaN         NaN   
reaction_time                -0.402179     

Yes, it's the delay time in days, exactly as Cohen described.

OK - let's use that instead:

In [25]:
def td_preprocessor(df):
    """ Process dataframe for TD trial types """
    onset_new, duration_orig, trial_type, delay, response, rt, onset_orig = [
        df[name].copy() for name in
        ['onset', 'duration', 'trial_type',
         'delay_time_days',
         'response_button',
         'reaction_time',
         'onset_orig']]
    # onset_orig has more decimal places, use that instead.  Rounding might in
    # fact be truncation, hence the 0.1 slippage here.
    assert np.max(np.abs(onset_new - onset_orig.round(2))) <= 0.1
    onset_orig.name = 'onset'
    # Use reaction times for duration, for most trials
    rt.name = 'duration'
    # Make responses into numbers
    response = response.map(TD_RESPONSE_MAP)
    tt = trial_type.copy()
    # Use reaction time for duration
    tt[trial_type == 'easy_par'] = 'easy'
    tt[trial_type == 'hard_par'] = 'hard'
    is_missed = response.isnull()
    tt[is_missed] = 'missed'
    # Use original duration (not RT) for missed trials
    rt[is_missed] = duration_orig[is_missed]
    amplitude = pd.Series(np.ones(len(df)), name='amplitude')
    main_trials = pd.concat([tt, onset_orig, rt, amplitude], axis=1)
    # Add parametric trial types
    extras = []
    for t_type in ('easy', 'hard'):
        trial_selector = tt == t_type
        amp_extra = delay[trial_selector].copy()
        amp_extra = amp_extra - amp_extra.mean()
        amp_extra.name = 'amplitude'
        tt_extra = tt[trial_selector].copy()
        tt_extra[:] = t_type + 'par'
        extras.append(pd.concat([tt_extra,
                                 onset_orig[trial_selector],
                                 rt[trial_selector],
                                 amp_extra],
                                axis=1))
    # Put the new trials at the end
    return pd.concat([main_trials] + extras, axis=0, ignore_index=True)

In [26]:
ours2 = td_preprocessor(df2)
ours2

trial_type     onset  duration   amplitude
0         easy   16.0120    1.4200    1.000000
1         easy   21.2562    2.6965    1.000000
2         easy   31.7613    1.8327    1.000000
3         easy   37.0055    1.2259    1.000000
4         hard   42.2665    1.3817    1.000000
5         hard   47.5108    1.3449    1.000000
6         easy   52.7551    1.2055    1.000000
7         easy   58.0159    1.7193    1.000000
8         hard   63.2602    2.4613    1.000000
9         hard   68.5044    1.2472    1.000000
10        easy   73.7654    1.0420    1.000000
11        hard   79.0096    2.3756    1.000000
12        easy   84.2539    1.0329    1.000000
13        hard   89.5147    2.1665    1.000000
14        easy   94.7590    1.9792    1.000000
15        hard  105.2641    1.0478    1.000000
16        hard  110.5084    1.2852    1.000000
17        hard  115.7526    1.5847    1.000000
18        easy  121.0135    1.9733    1.000000
19        hard  126.2579    1.4449    1.000000
20        hard  131.5021    0.8345    1.000000
21        hard  136.7629    1.6846    1.000000
22        hard  142.0073    1.0194    1.000000
23        hard  147.2515    0.7172    1.000000
24        hard  152.5124    1.6068    1.000000
25        hard  157.7566    0.8284    1.000000
26        easy  163.0009    1.6713    1.000000
27        easy  168.2617    1.2985    1.000000
28        easy  178.7669    0.9201    1.000000
29        easy  184.0112    0.9003    1.000000
..         ...       ...       ...         ...
162    hardpar  205.0048    0.8659  -93.833333
163    hardpar  210.2658    1.4937  269.166667
164    hardpar  215.5100    0.7626  -93.833333
165    hardpar  220.7542    0.8211  -74.833333
166    hardpar  231.2594    0.9479  -35.833333
167    hardpar  241.7646    2.0755   24.166667
168    hardpar  257.5140    1.3192  -88.833333
169    hardpar  262.7583    0.7531  -93.833333
170    hardpar  268.0024    0.9170  -88.833333
171    hardpar  273.2634    0.8076  -88.833333
172    hardpar  278.5077    0.6037  -93.833333
173    hardpar  283.7519    1.1436  -35.833333
174    hardpar  299.5013    0.7253  -74.833333
175    hardpar  304.7623    1.1524  269.166667
176    hardpar  310.0065    0.7695  -35.833333
177    hardpar  336.2610    0.8625  -74.833333
178    hardpar  341.5052    0.9426   24.166667
179    hardpar  346.7662    0.6669  -35.833333
180    hardpar  367.7598    0.7301  -35.833333
181    hardpar  383.5093    1.0981   24.166667
182    hardpar  404.5030    1.5759  269.166667
183    hardpar  409.7638    0.9881   24.166667
184    hardpar  451.7512    1.1061   24.166667
185    hardpar  472.7615    3.4302  -93.833333
186    hardpar  493.7551    0.8297  -35.833333
187    hardpar  499.0161    0.5352  -88.833333
188    hardpar  509.5046    0.6789  -74.833333
189    hardpar  514.7654    0.7155  -35.833333
190    hardpar  530.5149    0.6068  -93.833333
191    hardpar  556.7527    0.9506  269.166667

[192 rows x 4 columns]

Are we getting the same values as the parametric values recorded in this `.tsv`?

In [27]:
ours2[ours2['trial_type'] == 'hardpar']

trial_type     onset  duration   amplitude
144    hardpar   42.2665    1.3817  -93.833333
145    hardpar   47.5108    1.3449  -74.833333
146    hardpar   63.2602    2.4613  269.166667
147    hardpar   68.5044    1.2472  -74.833333
148    hardpar   79.0096    2.3756  -88.833333
149    hardpar   89.5147    2.1665  269.166667
150    hardpar  105.2641    1.0478  269.166667
151    hardpar  110.5084    1.2852  -74.833333
152    hardpar  115.7526    1.5847  -88.833333
153    hardpar  126.2579    1.4449  -88.833333
154    hardpar  131.5021    0.8345  -74.833333
155    hardpar  136.7629    1.6846  -35.833333
156    hardpar  142.0073    1.0194  -93.833333
157    hardpar  147.2515    0.7172  -88.833333
158    hardpar  152.5124    1.6068   24.166667
159    hardpar  157.7566    0.8284  269.166667
160    hardpar  189.2555    0.6815   24.166667
161    hardpar  199.7606    0.8720   24.166667
162    hardpar  205.0048    0.8659  -93.833333
163    hardpar  210.2658    1.4937  269.166667
164    hardpar  215.5100    0.7626  -93.833333
165    hardpar  220.7542    0.8211  -74.833333
166    hardpar  231.2594    0.9479  -35.833333
167    hardpar  241.7646    2.0755   24.166667
168    hardpar  257.5140    1.3192  -88.833333
169    hardpar  262.7583    0.7531  -93.833333
170    hardpar  268.0024    0.9170  -88.833333
171    hardpar  273.2634    0.8076  -88.833333
172    hardpar  278.5077    0.6037  -93.833333
173    hardpar  283.7519    1.1436  -35.833333
174    hardpar  299.5013    0.7253  -74.833333
175    hardpar  304.7623    1.1524  269.166667
176    hardpar  310.0065    0.7695  -35.833333
177    hardpar  336.2610    0.8625  -74.833333
178    hardpar  341.5052    0.9426   24.166667
179    hardpar  346.7662    0.6669  -35.833333
180    hardpar  367.7598    0.7301  -35.833333
181    hardpar  383.5093    1.0981   24.166667
182    hardpar  404.5030    1.5759  269.166667
183    hardpar  409.7638    0.9881   24.166667
184    hardpar  451.7512    1.1061   24.166667
185    hardpar  472.7615    3.4302  -93.833333
186    hardpar  493.7551    0.8297  -35.833333
187    hardpar  499.0161    0.5352  -88.833333
188    hardpar  509.5046    0.6789  -74.833333
189    hardpar  514.7654    0.7155  -35.833333
190    hardpar  530.5149    0.6068  -93.833333
191    hardpar  556.7527    0.9506  269.166667

In [28]:
df2[df2['trial_type'] == 'hard_par']['hard_par_orig']

4     -93.8333
5     -74.8333
8     269.1667
9     -74.8333
11    -88.8333
13    269.1667
15    269.1667
16    -74.8333
17    -88.8333
19    -88.8333
20    -74.8333
21    -35.8333
22    -93.8333
23    -88.8333
24     24.1667
25    269.1667
30     24.1667
32     24.1667
33    -93.8333
34    269.1667
35    -93.8333
36    -74.8333
38    -35.8333
40     24.1667
42    -88.8333
43    -93.8333
44    -88.8333
45    -88.8333
46    -93.8333
47    -35.8333
50    -74.8333
51    269.1667
52    -35.8333
56    -74.8333
57     24.1667
58    -35.8333
62    -35.8333
65     24.1667
68    269.1667
69     24.1667
77     24.1667
80    -93.8333
84    -35.8333
85    -88.8333
87    -74.8333
88    -35.8333
91    -93.8333
95    269.1667
Name: hard_par_orig, dtype: object